janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/Scanner.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.keycolumnvalue.scan;
+
+import org.janusgraph.diskstorage.PermanentBackendException;
+import org.janusgraph.diskstorage.TemporaryBackendException;
+import org.janusgraph.diskstorage.keycolumnvalue.KeyColumnValueStore;
+
+import java.util.concurrent.BlockingQueue;
+
+import static org.janusgraph.diskstorage.keycolumnvalue.scan.StandardScannerExecutor.Row;
+
+/**
+ * Produces data to {@link BlockingQueue<Row>}
+ * for each key in {@link KeyColumnValueStore}
+ *
+ * @author Sergii Karpenko (sergiy.karpenko@gmail.com)
+ */
+abstract class Scanner {
```

Comment:
Is it possible to rename it to something else? Its name gives me an illusion that StandardScanner should be its subclass, which is not the case.

Reply:
Do you mean StandardScannerExecutor when saying StandardScanner ?

Reply:
From perspective all looks logical StandardScannerExecutor executes Scanner

Reply:
Nope, I mean org.janusgraph.diskstorage.keycolumnvalue.scan.StandardScanner

Reply:
I'd rather rename `StandardScanner` to `StandardScanManager` or `StandardScanProcessor` because it's main responsibility is to mange scan jobs rather then run scans 
Other option is to rename `Scanner` to `Collector`